controllers/provisioners/eks/helpers.go

```diff
@@ -110,6 +110,37 @@ func (ctx *EksInstanceGroupContext) GetRemovedTags(asgName string) []*autoscalin
 	return removal
 }
 
+func (ctx *EksInstanceGroupContext) UpdateScalingProcesses(asgName string) error {
+	var (
+		instanceGroup         = ctx.GetInstanceGroup()
+		configuration         = instanceGroup.GetEKSConfiguration()
+		state                 = ctx.GetDiscoveredState()
+		scalingGroup          = state.GetScalingGroup()
+		specSuspendProcesses  = configuration.GetSuspendProcesses()
+		groupSuspendProcesses []string
+	)
+
+	for _, element := range scalingGroup.SuspendedProcesses {
+		groupSuspendProcesses = append(groupSuspendProcesses, *element.ProcessName)
+	}
+
+	processesToSuspend := common.Difference(specSuspendProcesses, groupSuspendProcesses)
+	err := ctx.AwsWorker.SetSuspendProcesses(asgName, processesToSuspend)
+	if err != nil {
+		return err
+	}
+	ctx.Log.Info("suspended scaling processes", "instancegroup", instanceGroup.GetName(), "scalinggroup", asgName, "difference", processesToSuspend)
```

Comment:
last thing! 💯 

Can we change it so it only logs when a change actually happens?
for example:
```
if suspend := common.Difference(specSuspendProcesses, groupSuspendProcesses); len(suspend) > 0 {
  if err := ctx.AwsWorker.SetSuspendProcesses(asgName, processesToSuspend); err != nil {
    return err
  }
  ctx.Log.Info("suspended scaling processes", "instancegroup", instanceGroup.GetName(), "scalinggroup", asgName, "processes", suspend)
}
```
same for resume - this would mean we can actually remove the len check from `AwsWorker.SetSuspendProcesses` into here.
also, can we change the key "difference" in the log to "processes" - it might be more accurate.

Sorry for the long review process!

Reply:
Hi @eytan-avisror ,
Absolutely no problem, please do let me know if there are any other improvements that you would be interested in.

I have made the above change, so it logs only when there is a change. The len checks in AwsWorker.SetSuspendProcesses and AwsWorker.SetResumeProcesses have been moved to the helper.

Thank you.